### ISSUE TITLE
Correlate Mechanisms of Resistance with Morphology Profiles

### ISSUE THREAD

**Issue by gwaygenomics** [broadinstitute/profiling-resistance-mechanisms#49]
@mekelley asked via an email thread what I am copying and pasting below (also noting here that I received permission to do so 😸)

---

Can identify the dominant morphological features that contributed to the 0.1% DMSO treated clones A and E from our earlier data (without the Costes features) and see if they’re the same dominant features from the 0.1% DMSO treated bortezomib resistant clones from group 3 (our recent data)?

What about dominant features from bortezomib resistant clones in the other groups from batch 3 (again without Costes features)?

The reason for these questions is to begin to correlate mechanisms of resistance (i.e. mutation in the target protein, PSMB5, or multidrug resistance or something else) with morphological profiles.

**Comment by gwaygenomics** [comment]
Thanks for the questions @mekelley, I will follow up below:

> Can identify the dominant morphological features that contributed to the 0.1% DMSO treated clones A and E from our earlier data (without the Costes features) and see if they’re the same dominant features from the 0.1% DMSO treated bortezomib resistant clones from group 3 (our recent data)?

> What about dominant features from bortezomib resistant clones in the other groups from batch 3 (again without Costes features)?

Yeah! This would be a cool comparison. Can you clarify you mean by the groups? Does this relate to the substructure we saw in the heatmaps?

> The reason for these questions is to begin to correlate mechanisms of resistance (i.e. mutation in the target protein, PSMB5, or multidrug resistance or something else) with morphological profiles.

Totally agree 😎 let's do it!

**Comment by gwaygenomics** [comment]
cc @ayberman

**Comment by ayberman** [comment]
> Thanks for the questions @mekelley, I will follow up below:
> 
> > Can identify the dominant morphological features that contributed to the 0.1% DMSO treated clones A and E from our earlier data (without the Costes features) and see if they’re the same dominant features from the 0.1% DMSO treated bortezomib resistant clones from group 3 (our recent data)?
> 
> > What about dominant features from bortezomib resistant clones in the other groups from batch 3 (again without Costes features)?
> 
> Yeah! This would be a cool comparison. Can you clarify you mean by the groups? Does this relate to the substructure we saw in the heatmaps?
> 
> > The reason for these questions is to begin to correlate mechanisms of resistance (i.e. mutation in the target protein, PSMB5, or multidrug resistance or something else) with morphological profiles.
> 
> Totally agree 😎 let's do it!

**Comment by ayberman** [comment]
> Thanks for the questions @mekelley, I will follow up below:
> 
> > Can identify the dominant morphological features that contributed to the 0.1% DMSO treated clones A and E from our earlier data (without the Costes features) and see if they’re the same dominant features from the 0.1% DMSO treated bortezomib resistant clones from group 3 (our recent data)?
> 
> > What about dominant features from bortezomib resistant clones in the other groups from batch 3 (again without Costes features)?
> 
> Yeah! This would be a cool comparison. Can you clarify you mean by the groups? Does this relate to the substructure we saw in the heatmaps?
>
Yes, we are referring to the 4 "groups" within the heatmap. We want to see if any of the 4 "groups" have similar feature signatures to clones A and E that we used for our initial studies. 

Clones A and E were genome sequenced and were confirmed to harbor a mutation in PSMB5. If any of the feature signatures for the 4 "groups" from the heatmap are similar to the feature signatures for clones A and E, we can hypothesize that maybe that group also has compromised PSMB5. 

If not, then the bortezomib resistance either 1. occurs via a different mechanism, or 2. is still induced by a PSMB5 mutation, but the features that result from this mutation are not defined or are influenced by other factors.

This could help us think about the data we have in terms of preparing it for publication!

> > The reason for these questions is to begin to correlate mechanisms of resistance (i.e. mutation in the target protein, PSMB5, or multidrug resistance or something else) with morphological profiles.
> 
> Totally agree 😎 let's do it!

**Comment by gwaygenomics** [comment]
I'm in the early stages of implementing this analysis, and I am doing a couple checks to see which approaches are likely to succeed. 

Initially, I thought a clustering-based approach might be helpful. We could use a simple clustering algorithm (like k-means) on concatenated data (batch 5, 6, 7 concatenated with batch 1 and 2) and see which clusters clone A and E land in. I like this approach because it reduces multiple testing burden, and we can assign confidence scores (based on multiple k-means initializations). However, I observed batch effect between the two data sources (see #53).

Therefore, unless we explore alternative batch effect mitigation strategies, any clustering solution will not be reliable. 

An alternative approach, which is similar to the analysis outlined in #48, can be to build a **consensus signature** of morphology features that are higher _and_ lower comparing cloneA + cloneE to the wildtype parental lines. We can also fit a random effects model to filter features that are impacted by batch, plate, or well effects. In this sense, we can get a ranked list of up and down morphology features. We can use this ranked list and apply a method like `ssGSEA` ([Barbie et al. 2009](https://doi.org/10.1038/nature08460) or `singscore` ([Foroutan et al 2018](https://doi.org/10.1186/s12859-018-2435-4)) to individual morphology profiles. We have lots of replicates of multiple resistant and wildtype clones, plus wildtype parental lines that we can assess. If we consistently see higher scores for particular resistant clones, but not for others, then we can have some reasonable expectation that the particular clone has compromised PSMB5. 

One question - do we know, _a priori_, the prevalence of compromised PSMB5 as resistance mechanism? Is it likely that all four resistant clones have it? Also just realizing now that we can apply this approach to all the other clones we have access to as well!

How does this sound? cc @ayberman @mekelley

**Comment by gwaygenomics** [comment]
Note - I applied `singscore` in #57 and present results in #58
